mlir/lib/Dialect/Rock/Transforms/AlignTiling.cpp

```diff
@@ -435,16 +558,15 @@ LogicalResult MILARewritePattern::matchAndRewrite(linalg::GenericOp laGeneric,
   }
 
   SmallVector<AffineMap> idxMaps = laGeneric.getIndexingMapsArray();
-  // Output must be indexed by identity map for this to work
```

Comment:
Huh. Is the thing you're doing with the inverses making this no longer the case?

If so, I still think there's a condition here. I saw `inversePermutation()` called in earlier code, so I'd like this if statement to check that the output index map is a permutation. And that it is equal to the map on the gemm output/corresponding la.generic input.

Reply:
So the check was too restrictive, IMO.

Lets see what are doing here : 

We are trying to fuse a subsequent kernel to the lowest level vector operation.
So there are three coordinate spaces in the question : input space, operator space, and output space.

The Linalg generic define the indexing maps as operator space --> input space and operator space --> output space.
So the overly restrictive logic here was that we assume operator space --> output space (i.e. which is what "Output must be indexed by identity map for this to work" is supposed to mean, I think) is an identity --> so that we can treat : 

operator space --> input space == output space --> input space.

This is what I felt as overly restrictive and we dont need that.

So the generic way to support that is : 

We invert the fold and compose : invert(op --> output) x (op --> input) == output --> input space which is what we are after. Therefore, we can properly ignore the variety of ways the linalg body could define the operation rather than being special cased here. All the indexing maps has to be invertible and that is guranteed by the construction of linalg generic for shape-to-loops hooks and vice-versa hooks to work.

Right, having said all that, I think if we still need conditions it should be in the form of iterator types and cyclic usage of output space within the op.

Reply:
Yes, it was intentionally strict to handle the simple case and this is a much better generalization. Thanks @manupak !